src/ostree/ot-builtin-size-summary.c

```diff
@@ -98,26 +99,27 @@ ostree_builtin_size_summary (int argc, char **argv, GCancellable *cancellable, G
       goto out;
     }
 
-  if (ostree_repo_get_commit_sizes (repo, revision,
-                                    &new_archived,
-                                    &new_unpacked,
-                                    &fetch_needed,
-                                    &archived, &unpacked,
-                                    &entries,
-                                    cancellable, error))
-    {
-      g_print ("Summary for refspec %s:\n"
-               "  files: %" G_GSIZE_FORMAT "/%" G_GSIZE_FORMAT " entries\n"
-               "  archived: %" G_GINT64_FORMAT "/%" G_GINT64_FORMAT "\n"
-               "  unpacked: %" G_GINT64_FORMAT "/%" G_GINT64_FORMAT "\n",
-               refspec,
-               entries - fetch_needed, entries,
-               archived - new_archived, archived,
-               unpacked - new_unpacked, unpacked);
-    }
+  if (!ostree_repo_get_commit_sizes (repo, revision,
+                                     &new_archived,
+                                     &new_unpacked,
+                                     &fetch_needed,
+                                     &archived, &unpacked,
+                                     &entries,
+                                     cancellable, error))
+    goto out;
+
+  g_print ("Summary for refspec %s:\n"
+           "  files: %" G_GSIZE_FORMAT "/%" G_GSIZE_FORMAT " entries\n"
+           "  archived: %" G_GINT64_FORMAT "/%" G_GINT64_FORMAT "\n"
+           "  unpacked: %" G_GINT64_FORMAT "/%" G_GINT64_FORMAT "\n",
+           refspec,
+           entries - fetch_needed, entries,
+           archived - new_archived, archived,
+           unpacked - new_unpacked, unpacked);
 
+  ret = TRUE;
```

Comment:
Sorry, I should have commented this a bit more. If the sizes information is available, `entries` is the number of files from the commit that are present in the repo. Since you might have a commit without files or you might have only the commit and none of the objects, this isn't really an error. The real error is if the sizes data is not available at all in the commit metadata (at least the way the code is now).

Does that make sense?